src/services/server.ts

```diff
@@ -1,31 +1,53 @@
-import type { FastifyInstance, FastifyPluginCallback, RouteOptions } from 'fastify';
+import type { FastifyInstance, FastifyPluginCallback } from 'fastify';
 import type { Logger } from 'pino';
 import fastifyRoutes from '@fastify/routes';
+import fastifyOauth2Verify, { type FastifyAuth0VerifyOptions } from 'fastify-auth0-verify';
+import env from 'env-var';
+
+import notFoundHandler from '../utilities/fastify/plugins/notFoundHandler.js';
+import { makeFastify } from '../utilities/fastify/server.js';
 
-import { configureFastify } from './fastify.js';
 import healthcheckRoutes from './routes/healthcheck.routes.js';
 import orgRoutes from './routes/org.routes.js';
-import memberRoutes from './routes/member.routes.js';
-import memberPublicKey from './routes/memberPublicKey.routes.js';
-import notFoundHandler from './plugins/notFoundHandler.js';
-import fastifyMongoose from './plugins/fastifyMongoose.js';
-
-const ROUTES: FastifyPluginCallback<RouteOptions>[] = [
-  healthcheckRoutes,
-  orgRoutes,
-  memberRoutes,
-  memberPublicKey,
-];
+import type { RouteOptions } from './types/RouteOptions.js';
+
+const ROOT_ROUTES: FastifyPluginCallback<RouteOptions>[] = [healthcheckRoutes, orgRoutes];
+
+function getOauth2PluginOptions(): FastifyAuth0VerifyOptions {
+  const audience = env.get('OAUTH2_TOKEN_AUDIENCE').required().asString();
+  const domain = env.get('OAUTH2_JWKS_URL').required().asUrlString();
+
+  const issuer = env.get('OAUTH2_TOKEN_ISSUER').asUrlString();
+  const issuerRegex = env.get('OAUTH2_TOKEN_ISSUER_REGEX').asRegExp('u');
+  if (
+    (issuer === undefined && issuerRegex === undefined) ||
+    (issuer !== undefined && issuerRegex !== undefined)
+  ) {
+    throw new Error('Either OAUTH2_TOKEN_ISSUER or OAUTH2_TOKEN_ISSUER_REGEX should be set');
+  }
+
+  return {
+    audience,
+    domain,
+    issuer: issuer ?? (issuerRegex as any),
+  };
+}
+
+async function makeApiServerPlugin(server: FastifyInstance): Promise<void> {
+  await server.register(fastifyRoutes);
+  await server.register(notFoundHandler);
+
+  const verifyOptions = getOauth2PluginOptions();
+  await server.register(fastifyOauth2Verify, verifyOptions);
+
+  await Promise.all(ROOT_ROUTES.map((route) => server.register(route)));
+}
 
 /**
  * Initialize a Fastify server instance.
  *
  * This function doesn't call .listen() so we can use .inject() for testing purposes.
  */
-export async function makeServer(logger?: Logger): Promise<FastifyInstance> {
-  return configureFastify(
-    [fastifyMongoose, fastifyRoutes, notFoundHandler, ...ROUTES],
-    undefined,
-    logger,
-  );
+export async function makeServer(customLogger?: Logger): Promise<FastifyInstance> {
```

Comment:
Do we need the logger as a parameter in this function?

Reply:
We do, when we need to check the logs in the unit tests